src/containers/profile/Profile.js

```diff
@@ -1,27 +1,55 @@
 import React, { useState, useEffect } from "react";
 import GithubProfileCard from "../../components/githubProfileCard/GithubProfileCard";
-import axios from "axios";
+import ApolloClient from "apollo-boost";
+import { gql } from "apollo-boost";
 import { openSource } from "../../portfolio";
 import Contact from "../contact/Contact";
 
-
 export default function Profile() {
   const [prof, setrepo] = useState([]);
   function setProfileFunction(array) {
     setrepo(array);
   }
   function getProfileData() {
-    const client=axios.get("https://api.github.com/users/"+openSource.githubUserName)
-      .then(function(response){
-            setProfileFunction(response.data);
+    const client = new ApolloClient({
+      uri: "https://api.github.com/graphql",
+      request: (operation) => {
+        operation.setContext({
+          headers: {
+            authorization: `Bearer ${openSource.githubConvertedToken}`,
+          },
+        });
+      },
     });
+
+    client
+      .query({
+        query: gql`
+      {
+        user(login:"${openSource.githubUserName}") { 
+          name
+          bio
+          isHireable
+          avatarUrl
+          location
+        }
+    }
+      `,
+      })
+      .then((result) => {
+        setProfileFunction(result.data.user);
+      });
   }
   useEffect(() => {
-    getProfileData();
+
+    if (openSource.showGithubProfile === "true") {
```

Comment:
Do we really need to check this condition?

Reply:
Yeah if someone has set false in showGithubProfile to false we won't call API

Reply:
We, already have a condition check-in line 2
https://github.com/saadpasta/developerFolio/blob/772d4cc9814a97db9b67044581868b240d0f21de/src/containers/profile/Profile.js#L22

Reply:
That's for the component not for the API call

Reply:
It would be good to log this in the console.